src/file.cpp

```diff
@@ -1999,10 +2003,10 @@ typedef struct _FILE_ALLOCATED_RANGE_BUFFER {
 		privs.PrivilegeCount = 1;
 		privs.Privileges[0].Attributes = SE_PRIVILEGE_ENABLED;
 
-		bool ret = pAdjustTokenPrivileges(token, FALSE, &privs, 0, NULL, NULL)
+		result = pAdjustTokenPrivileges(token, FALSE, &privs, 0, NULL, NULL)
 			&& GetLastError() == ERROR_SUCCESS;
```

Comment:
it looks a bit suspicious to not use a `bool` here for `result`. The resulting type of `operator&&` is `bool`, not `BOOL`. perhaps the result from `pAdjustTokenPrivileges()` should be checked against `TRUE` though.

Also, it seem reasonable to just return directly, rather than first storing the result in `result`, that would make the question of type a non-issue. the `master` branch is more strict with types as all implicit conversion warnings are enabled there

Reply:
Standard conformant. `bool` to `int` conversion is implicit, and `BOOL` stands for `int` in our case.

§4.7/4 from the C++ Standard says (Integral Conversion)

> If the source type is bool, the value false is converted to zero and the value true is converted to one.

Reply:
I'm not suggesting the code is invalid. I would just prefer to not rely on implicit conversions. That's why all conversion warnings have been enabled in `master`

Reply:
Can you confirm that there is a warning caused by this conversion? I supposed there shouldn't be any.

Reply:
https://clang.llvm.org/extra/clang-tidy/checks/readability-implicit-bool-conversion.html